drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1292 @@
+                 break;
+        case 12: data->status = VL53L1X_RANGESTATUS_XTALK_SIGNAL_FAIL; break;
+        case 13: data->status = VL53L1X_RANGESTATUS_MIN_RANGE_FAIL; break;
+        case 17: data->status = VL53L1X_RANGESTATUS_HARDWARE_FAIL; break;
+        case 18: data->status = VL53L1X_RANGESTATUS_SYNCRONISATION_INT; break;
+        case 19: data->status = VL53L1X_RANGESTATUS_RANGE_VALID_NO_WRAP_CHECK_FAIL; break;
+        default: data->status = VL53L1X_RANGESTATUS_NONE;
+    }
+
+    /* calibrate if necessary */
+    if (!dev->calibrated) {
+        _setup_manual_calibration(dev);
+        dev->calibrated = true;
+    }
+    #endif /* !MODULE_VL53L1X_BASIC */
+
+    EXEC_RET(_update_dss(dev, &raw));
+
+    /* calculate range */
+    uint16_t range;
+    range  = (uint16_t)raw.final_crosstalk_corrected_range_mm_sd0_hi << 8;
+    range |= raw.final_crosstalk_corrected_range_mm_sd0_lo;
+
+    /*
+     * apply correction gain
+     * gain factor of 2011 is tuning parm default
+     * (VL53L1_TUNINGPARM_LITE_RANGING_GAIN_FACTOR_DEFAULT)
+     * Basically, this appears to scale the result by 2011/2048, or about 98%
+     * (with the 1024 added for proper rounding).
+     */
+    data->distance = ((uint32_t)range * 2011 + 0x0400) / 0x0800;
+    /* Copyrighted POLOLU / ST API code end */
+
+    #if !MODULE_VL53L1X_BASIC
+    /* peak signal rate in 9.7 fixed point format */
+    data->signal_rate  = (uint32_t)raw.peak_signal_count_rate_mcps_sd0_hi << 8;
+    data->signal_rate |= raw.peak_signal_count_rate_mcps_sd0_lo;
+
+    /* ambient rate in in 9.7 fixed point format */
+    data->ambient_rate  = (uint32_t)raw.ambient_count_rate_mcps_sd0_hi << 8;
+    data->ambient_rate |= raw.ambient_count_rate_mcps_sd0_lo;
+
+    /* convert values from 9.7 to 16.16 */
+    data->signal_rate  <<= 9;
+    data->ambient_rate <<= 9;
+
+    #endif /* !MODULE_VL53L1X_BASIC */
+
+    /* restart measurement */
+    EXEC_RET(_start(dev));
+
+    #endif /* MODULE_VL53L1X_ST_API */
+
+    return VL53L1X_OK;
+}
+
+int vl53l1x_read_mm(vl53l1x_t *dev, int16_t *mm)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(mm != NULL);
+    DEBUG_DEV("mm=%p", dev, mm);
+
+    vl53l1x_data_t data;
+    EXEC_RET(vl53l1x_read_data (dev, &data));
+    *mm = data.distance;
+
+    return VL53L1X_OK;
+}
+
+int vl53l1x_int_config(vl53l1x_t *dev, vl53l1x_int_config_t* cfg,
+                                       void (*isr)(void *),
+                                       void *isr_arg)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(dev->params.pin_int != GPIO_UNDEF);
+
+    DEBUG_DEV("cfg=%p isr=%p isr_arg=%p", dev, cfg, isr, isr_arg);
+
+    if (!dev->int_init) {
+        dev->int_init = true;
+        gpio_init_int(dev->params.pin_int, GPIO_IN_PU, GPIO_FALLING,
+                      isr, isr_arg);
+    }
+
+    #if MODULE_VL53L1X_ST_API
+    ASSERT_PARAM(cfg != NULL);
+
+    EXEC_RET(VL53L1_StopMeasurement((VL53L1_Dev_t*)dev));
```

Comment:
```suggestion
    EXEC_RET(VL53L1_StopMeasurement(&dev->dev));
```

Reply:
Hm, `dev` is const in most functions which isn't a problem when calling ST API function where `dev` isn't const by casting it.